lib/demux/signal.rb

```diff
@@ -39,12 +48,13 @@ def payload_for(action)
       end
     end
 
-    def send(action)
+    def send(action, context: {})
       @demuxer.new(
         SignalAttributes.new(
           account_id: @account_id,
           action: String(action),
           object_id: @object_id,
+          context: @context.merge(context),
```

Comment:
by merging the context supplied directly into the context set on the class, we are giving precedence to the context supplied directly. For example, if you had `TestSignal.new(1, account_id: 9, context: { something: "foo" }).ping(something: "bar")` where the ping action then passed in that argument to send `send :ping, context: context` then `something` would be sent as "bar". That's not a primary use case, but it's a thing that is possible in this configuration. Passing in the context is mainly there for re-hydrating a signal from the transmission saved in the database before sending it.